### PR TITLE
chore(ui-react-core-notifications): add package boilerplate

### PR DESCRIPTION
**Pull request by sreeramsama**
#### Description of changes
Add package boilerplate

#### Checklist

- [x] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [x] PR description included
- [x] Relevant documentation is changed or added (and PR referenced)
- [x] `yarn test` passes and tests are updated/added
- [x] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/react-core-notifications/.estlintrc.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+module.exports = {
+  extends: ['amplify-ui/react'],
```
Comment (sreeramsama) [inline packages/react-core-notifications/.estlintrc.js:2]: Should we extend `amplify-ui/react-core` instead (which extends `amplify-ui/react`)?

File: packages/react-core-notifications/jest.config.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+module.exports = {
+  collectCoverage: true,
+  collectCoverageFrom: ['<rootDir>/src/**/*.(ts|tsx)'],
+  // ignore coverage for top level "export", version and style files
+  coveragePathIgnorePatterns: ['<rootDir>/src/(index|styles|version).(ts|tsx)'],
```
Comment (calebpollman) [inline packages/react-core-notifications/jest.config.js:5]: We shouldn't need to ignore "version" or "style" related files

**Review by calebpollman — COMMENTED** [review]
Great start! Thanks @sreeramsama

File: packages/react-core-notifications/.estlintrc.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+module.exports = {
+  extends: ['amplify-ui/react-core'],
```
[the remark this answers is not in the document]
Comment (sreeramsama) [reply packages/react-core-notifications/.estlintrc.js:2]: https://github.com/aws-amplify/amplify-ui/blob/main/packages/react-core/.eslintrc.js
It extends config of `amplify-ui/react`

File: packages/react-core-notifications/package.json [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    ".": {
+      "types": "./dist/types/index.d.ts",
+      "import": "./dist/esm/index.mjs",
+      "require": "./dist/index.js"
+    },
+    "./styles.css": "./dist/styles.css"
+  },
+  "browser": {
+    "./styles.css": "./dist/styles.css"
+  },
```
[the remark this answers is not in the document]
Comment (sreeramsama) [reply packages/react-core-notifications/package.json]: Removed style related dependencies and config

File: packages/react-core-notifications/.estlintrc.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (sreeramsama) [reply packages/react-core-notifications/.estlintrc.js:2]: Ah I see.
Are you referring to https://github.com/aws-amplify/amplify-ui/blob/main/packages/configs/eslint/configs/react.js?

[the remark this answers is not in the document]
Comment (calebpollman) [reply packages/react-core-notifications/.estlintrc.js:2]: That's correct. We only have two configs currently, `amplify-ui/react` and `amplify-ui/jest`

[the remark this answers is not in the document]
Comment (sreeramsama) [reply packages/react-core-notifications/.estlintrc.js]: Added an update to extend eslint config `amplify-ui/react`

**Review by 0618 — APPROVED** [review]

**Review by calebpollman — APPROVED** [review]

**Pull request merged**